### PR TITLE
Switched from AsyncDuplicateLock to AsyncKeyedLock

### PR DESCRIPTION
**Pull request by MarkCiliaVincenti**
Switched from AsyncDuplicateLock to AsyncKeyedLock, which provides better performance and lower memory usage.

### CODE REVIEW

**Review by f2calv — CHANGES_REQUESTED** [review]
I don't have time to test this (or your package) for a few days, but in the meantime;

- Please undelete AsyncDuplicateLock and add a deprecated attribute with a pointer to your library [Obsolete("???")] (it can be fully deleted in a future release)
- readonly AsyncKeyedLocker<string> _asyncKeyedLocker; should appear below ILogger (formatting is important to me)
- AsyncKeyedLock package should be moved from CasCap.Common.Threading.csproj and added to CasCap.Common.Caching.csproj
- thanks for the PR :)

**Comment by MarkCiliaVincenti** [comment]
> I don't have time to test this (or your package) for a few days, but in the meantime;
> 
> * Please undelete AsyncDuplicateLock and add a deprecated attribute with a pointer to your library [Obsolete("???")] (it can be fully deleted in a future release)
> * readonly AsyncKeyedLocker _asyncKeyedLocker; should appear below ILogger (formatting is important to me)
> * AsyncKeyedLock package should be moved from CasCap.Common.Threading.csproj and added to CasCap.Common.Caching.csproj
> * thanks for the PR :)

Done :)

File: src/CasCap.Common.Caching/Services/DistCacheService.cs [hunk 1]
```diff
@@ -33,11 +34,13 @@ public class DistCacheService : IDistCacheService
 
     public DistCacheService(ILogger<DistCacheService> logger,
         IOptions<CachingOptions> cachingOptions,
+        AsyncKeyedLocker<string> asyncKeyedLocker,
```
Comment (f2calv) [inline src/CasCap.Common.Caching/Services/DistCacheService.cs:36]: move directly below logger

File: src/CasCap.Common.Threading/Logic/AsyncDuplicateLock.cs [hunk 2]
```diff
@@ -3,6 +3,7 @@
 /// <summary>
 /// asynchronous locking based on a key
 /// </summary>
+[Obsolete("Use AsyncKeyedLock package instead.")]
```
Comment (f2calv) [inline src/CasCap.Common.Threading/Logic/AsyncDuplicateLock.cs]: Deprecated via use of AsyncKeyedLock, https://www.nuget.org/packages/AsyncKeyedLock

File: src/CasCap.Common.Caching/Services/DistCacheService.cs (hunk 1 above)
Comment (MarkCiliaVincenti) [reply src/CasCap.Common.Caching/Services/DistCacheService.cs:36]: Done.

File: src/CasCap.Common.Threading/Logic/AsyncDuplicateLock.cs (hunk 2 above)
Comment (MarkCiliaVincenti) [reply src/CasCap.Common.Threading/Logic/AsyncDuplicateLock.cs]: Done.

**Review by f2calv — APPROVED** [review]
thanks for the contribution :)

**Comment by MarkCiliaVincenti** [comment]
> thanks for the contribution :)

Thank you for your trust :) I also opened a similar PR @ https://github.com/f2calv/CasCap.GooglePhotosCli/pull/125

**Pull request merged**